test/PaymentRelay.test.ts

```diff
@@ -26,94 +27,241 @@ contract("PaymentRelay", (accounts) => {
       await paymentRelayContract.RECEIVER_ROLE(),
       receiver
     );
+    await paymentRelayContract.grantRole(
+      await paymentRelayContract.OPERATOR_ROLE(),
+      operator
+    );
   });
 
-  describe("As any user", function () {
-    it("I cannot interact with PaymentRelay directly", async function () {
-      try {
-        await paymentRelayContract.execPayment(sender, "100", "0x0", receiver, {
-          from: sender,
-        });
-        assert.fail("tokensReceived() did not throw.");
-      } catch (e: any) {
-        expect(e.message).to.includes("missing role");
-      }
-    });
-  });
-
-  describe("As any user", function () {
-    describe("When token are sent", function () {
-      const PAYMENT_UID = web3.utils.keccak256("PAYMENT_UID");
-      const amount = "1000";
+  describe.only("As any user", function () {
+    const PAYMENT_UID = web3.utils.keccak256("PAYMENT_UID");
+    const PAYMENT_UID_2 = web3.utils.keccak256("PAYMENT_UID_2");
+    const ANY_UID = web3.utils.keccak256("ANY_UID");
+    const amount = "500";
 
+    describe("When payment is reserved", function () {
```

Comment:
P2
Could be nice to check that we cannot execute nor refund when payment is not reserved.

Reply:
Agreed! The following tests should cover these exact situations:

- "Should block execution of payments that were not previously reserved"
- "Should block refund of payments that were not previously reserved"